packages/p2p/src/components/floating-rate/floating-rate.jsx

```diff
@@ -0,0 +1,114 @@
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import React from 'react';
+import { InputField, Text } from '@deriv/components';
+import { isMobile } from '@deriv/shared';
+import { localize } from 'Components/i18next';
+import { useStores } from 'Stores';
+import './floating-rate.scss';
+
+const FloatingRate = ({
+    change_handler,
+    className,
+    exchange_rate,
+    error_messages,
+    fiat_currency,
+    local_currency,
+    offset,
+    placeholder,
+    ...props
+}) => {
+    const { general_store } = useStores();
+    const { name, value, required } = props;
+    const market_feed = value ? parseFloat(exchange_rate * (1 + value / 100)).toFixed(2) : exchange_rate;
+
+    return (
+        <div className={classNames(className, 'floating-rate')}>
+            <section className={classNames('floating-rate__field', { 'mobile-layout': isMobile() })}>
+                <InputField
+                    ariaLabel='Floating rate'
+                    classNameInlinePrefix='floating-rate__percent'
+                    classNameInput={classNames('floating-rate__input', {
+                        'floating-rate__input__error_field': error_messages,
+                    })}
+                    decimal_point_change={2}
+                    id='floating_rate_input'
+                    inline_prefix='%'
+                    is_autocomplete_disabled
+                    is_float
+                    is_incrementable
+                    is_signed
+                    inputmode='decimal'
+                    increment_button_type='button'
+                    max_value={offset.upper_limit}
+                    min_value={offset.lower_limit}
+                    name={name}
+                    onChange={change_handler}
+                    placeholder={placeholder}
+                    setCurrentFocus={general_store.setCurrentFocus}
+                    required={required}
+                    type='number'
+                    value={value}
+                />
+                <div className='floating-rate__mkt-rate'>
+                    <Text
+                        as='span'
+                        size='xxxs'
+                        color='prominent'
+                        weight='normal'
+                        line_height='xxs'
+                        className='floating-rate__mkt-rate__label'
```

Comment:
Same here. `block__element1__element2` is not a valid BEM selector.

Reply:
I will refer to BEM and make the changes